src/persist/src/indexed/mod.rs

```diff
@@ -670,27 +718,24 @@ impl<L: Log, B: Blob> Indexed<L, B> {
     /// assuming data mostly arrives in order, or not very far in advance of the
     /// currently sealed time. We will need to revisit the unsealed compaction if
     /// that assumption stops being true.
-    fn compact(&mut self) -> Result<(), Error> {
+    fn compact_unsealed(&mut self) -> Result<(), Error> {
         // NB: This validate_pending_empty is intentionally a returned error
         // instead of an assert because it's a precondition (and so a violation
         // means a usage error by the caller of this).
         self.validate_pending_empty()?;
 
         let compaction_start = Instant::now();
-        let ret = self.apply_unbatched_cmd(|state, _, maintainer| state.compact_inner(maintainer));
+        let ret = self.apply_unbatched_cmd(|state, _| state.compact_unsealed());
 
         // Track compaction_seconds even if compaction failed.
         self.metrics
             .compaction_seconds
```

Comment:
ping?

Reply:
yes sorry I thought I had responded to this earlier -- we don't have access to Metrics in maintainer - im going to follow up with you because i wasn't sure if there was an intentional reason Maintainer didn't have metrics or if having an Arc<Metrics> as a field seemed totally fine?

Reply:
no reason. it just hadn't need them yet, so I didn't bother doing the plumbing

Reply:
done.